src/inmanta/deploy/timers.py

```diff
@@ -266,28 +304,26 @@ def _setup_deploy(deploy_interval: int) -> tuple[int, str, TaskPriority]:
 
             if self.periodic_repair_interval and self.periodic_deploy_interval:
                 if self.periodic_repair_interval < self.periodic_deploy_interval:
-                    countdown, reason, priority = _setup_repair(self.periodic_repair_interval)
+                    _setup_repair(self.periodic_repair_interval)
                 else:
-                    countdown, reason, priority = _setup_deploy(self.periodic_deploy_interval)
+                    _setup_deploy(self.periodic_deploy_interval)
             elif self.periodic_repair_interval:
-                countdown, reason, priority = _setup_repair(self.periodic_repair_interval)
+                _setup_repair(self.periodic_repair_interval)
             else:
                 assert self.periodic_deploy_interval is not None  # mypy
-                countdown, reason, priority = _setup_deploy(self.periodic_deploy_interval)
+                _setup_deploy(self.periodic_deploy_interval)
 
-        self.resource_timers[resource].cancel()
-        self.resource_timers[resource].set_timer(
-            countdown=countdown,
-            reason=reason,
-            priority=priority,
-        )
-
-    def update_timers(self, resources: Collection[ResourceIdStr], *, are_compliant: bool) -> None:
+    def update_timers(self, resources: Collection[ResourceIdStr]) -> None:
         """
         Make sure the given resources are marked for eventual re-deployment in the future.
+
+        Must be called under scheduler lock
+        Should not be called with resources that are already scheduled
+         -> this implies that every resource is expected to have a last_deployed time already,
+            otherwise it would be in the queue
```

Comment:
I'm not sure this is True for transitive_unblocked resources that were never deployed before ? https://github.com/inmanta/inmanta-core/blob/fee8e60eff44ebcc7e85278c3fa69dad24bf5466/src/inmanta/deploy/scheduler.py#L923

Reply:
I would assume those to be in the dirty set?